metaswitch/common/mib.py

```diff
@@ -40,10 +40,13 @@ def get_all_stats(self, columns):
         """
         # Add INDEX to the list of fields to retrieve and parse - we
         # need this in order to check whether a node is a statistic or not.
-        if "INDEX" not in columns:
-            columns.append("INDEX")
+        # To do this we use a copy of columns to keep ownership of columns in
+        # the calling script.
+        columns_copy = columns[:]
+        if "INDEX" not in columns_copy:
+            columns_copy.append("INDEX")

```

Comment:
Oops, I should have spotted this sooner.

This code's in the wrong place:
* The `Statistic` object's method `is_index_field` relies on the `"INDEX"` column being available.
* That means that it's an error to create a `Statistic` object without the `"INDEX"` column.
* So it shouldn't be possible to do so.

But this code allows us to create a `Statistic` object without the `"INDEX"` field.